exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Table.kt

```diff
@@ -1380,28 +1363,54 @@ open class Table(name: String = "") : ColumnSet(), DdlAware {
      */
     @JvmName("transformNullable")
     fun <Unwrapped : Any, Wrapped : Any> Column<Unwrapped?>.transform(
-        transformer: ColumnTransformer<Unwrapped, Wrapped>
-    ): Column<Wrapped?> = transform(ColumnWithTransform(columnType, transformer))
+        transformer: ColumnTransformer<Unwrapped?, Wrapped?>
+    ): Column<Wrapped?> {
+        val newColumn = copyWithAnotherColumnType<Wrapped?>(NullableColumnWithTransform(this.columnType, transformer)) {
+            defaultValueFun = this@transform.defaultValueFun?.let { { it()?.let { value -> transformer.wrap(value) } } }
+        }
+        return replaceColumn(this, newColumn)
+    }
 
     /**
-     * Applies the transformation column type to the nullable column.
+     * `nullTransform` is a special transformation that allows converting a non-nullable column
+     * into a nullable one.
+     *
+     * This transformation doesn't affect DDL generation, but it enables reflecting non-null values
+     * from the database as `null` in Kotlin (e.g., converting an empty string from a non-nullable text column,
+     * empty lists, negative IDs, etc., to `null`).
```

Comment:
I would consider rephrasing this because (even with the 2nd sentence) it still sounds like the column is literally being changed into a nullable one on the database side, and also some people may not fully get what DDL means.
I would pull some of what you state in the 2nd line to the 1st, so there's no room for confusion, for example something like:
```kt
/**
  * Applies a special transformation that allows a non-nullable database column
  * to accept and/or return values as `null` on the client side.
  *
  * This transformation does not alter the column's definition in the database,
  * which will still be `NON NULL`. It enables reflecting non-null values
  * from the database as `null` in Kotlin (e.g., converting an empty string from a
  * non-nullable text column, empty lists, negative IDs, etc., to `null`).
  */
```